libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -402,15 +403,24 @@ public override async Task<ResourceResponse[]> SendActivitiesAsync(ITurnContext
                 {
                     // if it is a Trace activity we only send to the channel if it's the emulator.
                 }
-                else if (!string.IsNullOrWhiteSpace(activity.ReplyToId))
-                {
-                    var connectorClient = turnContext.TurnState.Get<IConnectorClient>();
-                    response = await connectorClient.Conversations.ReplyToActivityAsync(activity, cancellationToken).ConfigureAwait(false);
-                }
                 else
                 {
-                    var connectorClient = turnContext.TurnState.Get<IConnectorClient>();
-                    response = await connectorClient.Conversations.SendToConversationAsync(activity, cancellationToken).ConfigureAwait(false);
+                    if (!activity.ServiceUrl.StartsWith("http"))
+                    {
+                        // Streaming connection post-processing
+                        TokenResolver.CheckForOAuthCards(this, turnContext, activity, cancellationToken);
+                    }
+
+                    if (!string.IsNullOrWhiteSpace(activity.ReplyToId))
```

Comment:
Add comment: why? what does it mean that the activity has a non-empty ReplyToId?

Reply:
Sorry, if you look, I didn't change this, I simply indented it. I believe from protocol that ReplyToId is not required, but if present controls whether this is a reply or just sending to a conversation. Bu I will add that comment.